### DISCUSSION TITLE
The request signature we calculated does not match the signature you provided. Check your key and signing method.

### DISCUSSION THREAD

**Discussion by ZengTianShengZ** [aws/aws-sdk-js-v3#4467] [Q&A]
I made a page for uploading files, and there is no problem reporting files in the Chrome browser, but the upload fails in the [quark](https://www.myquark.cn/) browser, prompting `SignatureDoesNotMatch`.
I think it is caused by differences in browsers, but I don't know how to troubleshoot this problem. I don't understand the logic of the internal signature of @aws-sdk/client-s3 sdk, so how can I troubleshoot this problem.

request header information
```
{
    "amz-sdk-invocation-id":"a4c15d3d-ed94-4d7c-9b5f-2d6589709b26",
    "amz-sdk-request":"attempt=1; max=3",
    "authorization":"AWS4-HMAC-SHA256 Credential=STS.NT9cTJUXCztcnczCSu6xaWymG/20230224/oss-cn-hangzhou/s3/aws4_request, SignedHeaders=amz-sdk-invocation-id;amz-sdk-request;content-length;content-type;host;x-amz-content-sha256;x-amz-date;x-amz-security-token;x-amz-user-agent, Signature=f24cb417ea4ddfcca6afd6cf684dbbfdc90329ebb4281822f5c56428a5821dca",
    "content-type":"image/png",
    "x-amz-content-sha256":"UNSIGNED-PAYLOAD",
    "x-amz-date":"20230224T032508Z",
    "x-amz-security-token":"CAISpAR1q6Ft5B2yfSjIr5eMKO7+uIdizbaIbEXLp1MgOvdNuLzGpTz2IH1OeHBhAegfsPs1mGlX6vYclrh8R5JeREPCKMd567wJoVr4MtJHKSJRbeRW5qe+EE2/VjR6tq27OpdyJrGwU/OpbE++LUYRpZLxaTSlWXG8LJSNkuQJR98LXw6+H+gvZrJRPRAwh8IGEnHTOP2xUHvtmXGCNFd0nQB+hGhji8G42dbOxRfCq1f4x+QJxeH2OYP2V9RwG519XtypvopxbbGT7TZU7BFGzK5skM43/izc7PCbAkVQ6gmLFvfQ7tx3JRVlW7A+HKdIzov7juYqkevWj4D61xtoIOVJUjzWXpyd2M/DEe/LRtEibuTfOm/R982LKpjtuhk4XGofMwpGYLhRLWRrWzUmTSvGD7Kj5FHSax2/drCB16g6q3EcrTPS8MGNOkKETpiAzC8cIeVcVU4zMAMM1mGTfc11egdXIVR8HOTXH8chYxxTsqLnu0jeXyglwHhL3bf3bOiEvbsEO8etHMBJ3YcGZZtA9mcjShH5Qq7pyBlELTU0I/pf27K/PoSkuv3XgrecbePdC/8B/1lTdHXfpjbKCy0AdHSgoZ8BEQPGpZ2Alf2Wo8g7SFJzvo1lYFvbIYo3nTtP+LCv6hmfyujnX02x83AwktuipNQVtRI1Jq/93bDI4meB4CzOe9U1kcLNUWhhQBC6fnV/yf+fnHUDqgDoYpI4yI6OgRqAASbp7DyWOnl8wkstPCzA2zpIt7wmZnhgNSHwm9JxwXaxFuFI4JJR7gHTPskoKJlYYg1dIsX0YLeYVHTl36L1Ix9FWm1WWVJyVYXd1SE9TiMXuZI/aFGIJmiz2cF2rRo5eCJEKTktb8/+E4djmNk+ndyCxqtr089gIP6JpuON4Xr9",
    "x-amz-user-agent":"aws-sdk-js/3.137.0 os/iOS/16.3.1 lang/js md/browser/Safari_unknown api/s3/3.137.0"
}
```
  error message
```
{
    "name":"SignatureDoesNotMatch",
    "message":"file upload failed:::The request signature we calculated does not match the signature you provided. Check your key and signing method."
}
```

```
 "@aws-sdk/client-s3": "^3.137.0",
```

Error reporting address,Need to visit under [quark](https://www.myquark.cn/) browser to report an error
https://www.gaoding.com/ai

**Comment by yenfryherrerafeliz** [comment]
Hi @ZengTianShengZ, sorry you are having this issue. Could you please share a sample code that helps us understand what you are trying to do. Would it be also possible for you to test with the latest version of the SDK?

Thanks!

**Comment by ZengTianShengZ** [comment]
Hi @yenfryherrerafeliz ,sorry, I can't give an example that can be used directly, my token is dynamically obtained. You can use this [example : discussions_4467](https://github.com/ZengTianShengZ/aws-sdk-js-v3-demo/tree/discussions_4467), and then modify [it](https://github.com/ZengTianShengZ/aws-sdk-js-v3-demo/blob/bbd1d6f664e7261b80a5dfff8d8d836e2d9c5ba8/src/index.ts#L8) to your own config to reproduce.

[quark browser](https://www.myquark.cn/) will throw the following error.

<img src="https://user-images.githubusercontent.com/15622519/223044158-f779c96f-6eb0-4d89-b58b-6c08eba5da18.png" width="300">

```

{
"name": "SignatureDoesNotMatch",
"$fault": "client",
"$metadata": {
"httpStatusCode": 403,
"attempts": 1,
"totalRetryDelay": 0
},
"Code": "SignatureDoesNotMatch",
"RequestId": "6405909C7AA92E3539725D47",
"HostId": "gdesign-dam-fat.oss-cn-hangzhou.aliyuncs.com",
"AWSAccessKeyId": "STS.NTLv19pkeFyBh8agU4458Dphk",
"SignatureProvided": "b29ac69964647101dd54e785f03b659b06d208315d9d805d6d8daea4f26eaaa6",
"StringToSign": "AWS4-HMAC-SHA256\n20230306T070500Z\n20230306/oss-cn-hangzhou/s3/aws4_request\n3037d2e76180b1ed44dbd1390c0f3656c49479a65accb0bfdaba0023726de94d",
"StringToSignBytes": "41 57 53 34 2D 48 4D 41 43 2D 53 48 41 32 35 36 0A 32 30 32 33 30 33 30 36 54 30 37 30 35 30 30 5A 0A 32 30 32 33 30 33 30 36 2F 6F 73 73 2D 63 6E 2D 68 61 6E 67 7A 68 6F 75 2F 73 33 2F 61 77 73 34 5F 72 65 71 75 65 73 74 0A 33 30 33 37 64 32 65 37 36 31 38 30 62 31 65 64 34 34 64 62 64 31 33 39 30 63 30 66 33 36 35 36 63 34 39 34 37 39 61 36 35 61 63 63 62 30 62 66 64 61 62 61 30 30 32 33 37 32 36 64 65 39 34 64 ",
"EC": "0002-00000429",
"message": "The request signature we calculated does not match the signature you provided. Check your key and signing method."
}
```

but safari browser won't.

<img src="https://user-images.githubusercontent.com/15622519/223044675-dac31756-18ad-427e-aa5d-c102de8781e8.png" width="300">

**Comment by ZengTianShengZ** [comment]
Is there anyone who has had a similar problem who can help?

**Comment by RanVaknin** [comment]
Hi @ZengTianShengZ ,

This is most likely caused by the browser adding / modifying the request.
There is a discrepancy between the signature the SDK calculates and the actual request sent. This is common when using proxies as the proxy layer might overwrite or add some headers to the request that were not accounted for in the signature. 

You need to check the request payload and headers in Chrome and then compare it to the request made by Quark. 

Thanks,
Ran~
